src/utilities/unique-id/hooks.ts

```diff
@@ -0,0 +1,38 @@
+import {useContext, useRef} from 'react';
+import {UniqueIdFactoryContext} from './context';
+
+/**
+ * Returns a unique id that remains consistent across multiple rerenders of the
+ * same hook
+ * @param prefix Defines a prefix for the ID. You probably want to set this to
+ *   the name of the component you're calling `useUniqueId` in.
+ * @param overrideId Defines a fixed value to use instead of generating a unique
+ *   ID. Useful for components that allow consumers to specify a fixed ID.
+ */
+export function useUniqueId(prefix = '', overrideId = '') {
+  const idFactory = useContext(UniqueIdFactoryContext);
+
+  // By using a ref to store the uniqueId for each incovation of the hook and
+  // checking that it is not already populated we ensure that we don't generate
+  // a new ID on ever rerender of a component.
+  const uniqueIdRef = useRef<string | null>(null);
```

Comment:
That won't isolate the id value to be within a single component. That'd mean multiple components would end up picking up that same value. This would also break the ability to have unique counters per prefix.